src/config/rcxml.c

```diff
@@ -51,6 +55,38 @@ enum font_place {
 static void load_default_key_bindings(void);
 static void load_default_mouse_bindings(void);
 
+static void
+fill_windowswitcher_field(char *nodename, char *content)
+{
+	if (!strcasecmp(nodename, "field.fields.windowswitcher")) {
+		current_field = znew(*current_field);
+		wl_list_append(&rc.windowswitcher.fields, &current_field->link);
+		return;
+	}
+
+	string_truncate_at_pattern(nodename, ".field.fields.windowswitcher");
+	if (!content) {
+		/* intentionally left empty */
+	} else if (!current_field) {
+		wlr_log(WLR_ERROR, "no <field>");
+	} else if (!strcmp(nodename, "content")) {
+		if (!current_field->content) {
+			current_field->content = xstrdup(content);
+		}
+	} else if (!strcmp(nodename, "width") && !strchr(content, '%')) {
+		wlr_log(WLR_ERROR, "Removing invalid field '%s': %s='%s' misses"
+			" a trailing %%", current_field->content, nodename, content);
+		wl_list_remove(&current_field->link);
+		zfree(current_field->content);
+		zfree(current_field);
```

Comment:
`current_field = NULL`? Otherwise we'd have a dangling pointer around

Reply:
zfree() does that

Reply:
Oh, wow. That is what reviewing code while tired does.